src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/KestrelHttpParser.cs

```diff
@@ -66,6 +68,27 @@ public KestrelHttpParser(IKestrelTrace log)
                 span = startLineBuffer.ToSpan();
             }
 
+            // pathStart
+            // |    pathEnd and queryStart
```

Comment:
Only if path had been decoded in place and query not moved (e.g. path could get shorter) - though this doesn't currently happen